### PR TITLE
edk: upgrade to 20240524; hvloader: upgrade ekd2 version

### PR DESCRIPTION
**Pull request by BettyRain**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade edk2 to 20240524
Upgrade edk2 version in hvloader

###### Change Log  

- Upgrade edk2 to 20240524
- Upgrade edk2 version in hvloader
- Fix CVEs: CVE-2022-36763, CVE-2022-36764, CVE-2022-36765, CVE-2023-45230, CVE-2023-45232, 
    CVE-2023-45233, CVE-2023-45234, CVE-2023-45235, CVE-2023-45236, CVE-2023-45237

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-36763
- https://nvd.nist.gov/vuln/detail/CVE-2022-36764
- https://nvd.nist.gov/vuln/detail/CVE-2022-36765
- https://nvd.nist.gov/vuln/detail/CVE-2023-45230
- https://nvd.nist.gov/vuln/detail/CVE-2023-45232
- https://nvd.nist.gov/vuln/detail/CVE-2023-45233
- https://nvd.nist.gov/vuln/detail/CVE-2023-45234
- https://nvd.nist.gov/vuln/detail/CVE-2023-45235
- https://nvd.nist.gov/vuln/detail/CVE-2023-45236
- https://nvd.nist.gov/vuln/detail/CVE-2023-45237

###### Test Methodology

- Local build
- Buddy build pipeline build id: [618035](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=618035&view=results)

### CODE REVIEW

**Comment by Camelron** [comment]
Testing Kata scenario here: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=618363&view=results

**Review by Camelron — APPROVED** [review]
new hvloader successfully installed and passed basic boot tests for kata scenario.

File: SPECS/edk2/edk2.spec [hunk 1]
```diff
@@ -337,7 +347,9 @@ cp -a -- \
    %{SOURCE30} %{SOURCE31} %{SOURCE32} \
    %{SOURCE40} %{SOURCE41} %{SOURCE42} %{SOURCE43} %{SOURCE44} \
    %{SOURCE45} %{SOURCE46} %{SOURCE47} %{SOURCE48} \
-   %{SOURCE80} %{SOURCE81} %{SOURCE82} \
+   %{SOURCE50} \
+   %{SOURCE60} \
+   %{SOURCE80} %{SOURCE81} %{SOURCE82} %{SOURCE82} \
```
Comment (Camelron) [inline SPECS/edk2/edk2.spec:352]: shouldn't this have SOURCE83 at the end, not two SOURCE82s?

Comment (BettyRain) [reply SPECS/edk2/edk2.spec:352]: That's why we need PRs, thank you for noticing!

Comment (BettyRain) [reply SPECS/edk2/edk2.spec]: New buddy build after changing this:
[618523](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=618523&view=results)

**Review by Camelron — APPROVED** [review]

File: SPECS/edk2/edk2.spec [hunk 2]
```diff
@@ -115,6 +119,11 @@ Patch0013: 0013-UefiCpuPkg-MpInitLib-fix-apic-mode-for-cpu-hotplug.patch
 Patch0014: 0014-CryptoPkg-CrtLib-add-stat.h.patch
 Patch0015: 0015-CryptoPkg-CrtLib-add-access-open-read-write-close-sy.patch
 Patch0016: 0016-OvmfPkg-set-PcdVariableStoreSize-PcdMaxVolatileVaria.patch
+%if 0%{?fedora} >= 38 || 0%{?rhel} >= 10
```
Comment (dmcilvaney) [inline SPECS/edk2/edk2.spec:122]: Should we also do this for azl?

Comment (BettyRain) [reply SPECS/edk2/edk2.spec:122]: Nope, I'll delete it and the patch that they have for it! You are right

**Review by dmcilvaney — APPROVED** [review]

**Comment by BettyRain** [comment]
Adding some clarifications to not lose the track of thought later:
- I removed -p1 in hvloader.spec because I removed both patches in spec file, since the new version of edk2 has these CVEs solved in it. To check it I checked lines of the patch and lines of the code in the new version of edk2, verified that they were added in the new version and don't need to be added by the patches. Or verified that these files don't exist anymore. Depends on the situation. Also, the patches weren't applicable and were giving an error.
- Patch 17 in edk2.spec suppresses the warnings that we would get from r/w/e permissions on files. And we don't want to add azl in this "if" line, because we don't want to suppress the warnings in azl. Also we don't want to remove it because it will be harder to make changes next time. But I'll raise a PR to add %if NOT azl macro that will disable that area of code for azl, to keep track of it and show that we turn it off intentionally.

**Pull request merged**